_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -70,12 +89,54 @@ proc _cmake_get_build_dir {} {
     return [option worksrcpath]
 }
 
-default configure.dir {[_cmake_get_build_dir]}
-
-pre-configure {
-    file mkdir ${configure.dir}
+option_proc cmake.generator cmake::handle_generator
+proc cmake::handle_generator {option action args} {
+    global cmake.generator destroot destroot.target build.cmd build.post_args
+    global depends_build destroot.post_args build.jobs
+    if {${action} eq "set"} {
+        switch -glob ${args} {
```

Comment:
I'd still prefer if this switched over `[lindex $args 0]` only. You wanted to do this change after a squash, right before the merge, so now would be a good time?